src/libgap-api.c

```diff
@@ -0,0 +1,134 @@
+// LibGAP API - API for using GAP as shared library.
+
+#include "libgap-api.h"
+
+#include "bool.h"
+#include "calls.h"
+#include "gapstate.h"
+#include "gvars.h"
+#include "intobj.h"
+#include "opers.h"
+#include "plist.h"
+#include "stringobj.h"
+
+Obj FuncREAD_ALL_COMMANDS(
+    Obj self, Obj instream, Obj echo, Obj capture, Obj resultCallback);
+
+// Setup and initialisation
+
+void GAP_Initialize(int          argc,
+                    char **      argv,
+                    char **      env,
+                    CallbackFunc markBagsCallback,
+                    CallbackFunc errorCallback)
+{
+    InitializeGap(&argc, argv, env);
+    SetExtraMarkFuncBags(markBagsCallback);
+    STATE(JumpToCatchCallback) = errorCallback;
```

Comment:
So since we have `SetExtraMarkFuncBags`, I suppose we want `SetErrorCallback`?

Reply:
I would think that's preferable, yes (overall, I'd prefer to get rid of as many global (non-static) variables as possible.

Reply:
So, I misunderstood about this: this is *only* meant to be called by `GAP_Initialize`, so there actually is no need for a setter for `STATE(JumpToCatchCallback)`, at least for now.

I still think `SetExtraMarkFuncBags` was "correct", because its presence means we can avoid exposing local state of GASMAN.

In other words, it's fine by me if the above line stays as it is. Sorry for the confusion.